driver-core/src/main/com/mongodb/client/model/TimeSeriesOptions.java

```diff
@@ -104,12 +109,102 @@ public TimeSeriesOptions granularity(@Nullable final TimeSeriesGranularity granu
         return this;
     }
 
+    /**
+     * Returns the maximum time span between measurements in a bucket.
+     *
+     * @param timeUnit the time unit.
+     * @return time span between measurements, or {@code null} if not set.
+     * @since 4.10
+     * @mongodb.server.release 6.3
+     * @mongodb.driver.manual core/timeseries-collections/ Time-series collections
+     * @see #bucketMaxSpan(Long, TimeUnit)
+     */
+    @Nullable
+    public Long getBucketMaxSpan(final TimeUnit timeUnit) {
+        if (bucketMaxSpanSeconds == null) {
+            return null;
+        }
+        return timeUnit.convert(bucketMaxSpanSeconds, TimeUnit.SECONDS);
+    }
+
+    /**
+     * Sets the maximum time span between measurements in a bucket.
+     * <p>
+     * The value of {@code bucketMaxSpanSeconds} must be the same as {@code bucketRoundingSeconds}.
+     * If you set the {@code bucketMaxSpanSeconds}, parameter, you can't set the granularity parameter.
+     * </p>
+     *
+     * @param bucketMaxSpan time span between measurements. After conversion to seconds using {@link TimeUnit#convert(long, java.util.concurrent.TimeUnit)},
+     * the value must be &gt;= 1. {@code null} can be provided to unset any previously set value.
+     * @param timeUnit the time unit.
```

Comment:
I want to make sure I understand your request correctly. I believe, the statement `{@code null} can be provided to unset any previously set value` implies that if the argument for `bucketMaxSpan`or `bucketRounding` is set to null, then any previously set value will be unset (meaning that it returns to previous state when it is disregarded). It is implied that all unset values are ignored, it might be worth mentioning at the class level documentation. 

I guess for better clarify we could mention that "By default all unset values are ignored" and "{@code null} can be provided to unset any previously set value."

Let me know if you have any further suggestions

Reply:
I was talking about the value of the `timeUnit` parameter. This argument is ignored if `bucketMaxSpan` is `null`, i.e., when unsetting.

However, maybe it shouldn't be completely ignored, maybe the `null`-check still must be done, given that the parameter is not annotated with `@Nullable`? If we don't have the check, it is quite likely that the methods will be called as `bucketRounding(null, null)`.

Reply:
Oh, i misunderstood initially. Yes, i agree,  adding a null check for the timeUnit parameter is a good idea. Thank you for bringing it up